src/scc_hypervisor_collector/api/uploader.py

```diff
@@ -5,7 +5,59 @@
 collected from the specified backends to the SCC using the provided
 credentials.
 """
+import json
+import logging
+import gzip
+from typing import Dict
+from importlib_metadata import version
+import requests
+from requests.exceptions import RequestException
+from .configuration import SccCredsConfig
 
 
 class SCCUploader:
     """SCC Uploader for scc-hypervisor-collector."""
+
+    def __init__(self, scccredsconfig: SccCredsConfig):
+        """Initialiser for SCCUploader"""
+        self._log = logging.getLogger(__name__)
+
+        # save the parameters
+        self._scccredsconfig: SccCredsConfig = scccredsconfig
+        self.username = self._scccredsconfig.username
+        self.password = self._scccredsconfig.password
+
+        self.auth = requests.auth.HTTPBasicAuth(self.username, self.password)
+
+        self.headers = {
+            'X-Gatherer-Version':
+                version('virtual-host-gatherer'),
```

Comment:
See above suggestion for renaming the version function when importing to make these lines a little more self documenting...